### ISSUE TITLE
Add a function to torch.nested to create nested tensors from a buffer and sizes

### ISSUE THREAD

**Issue by fleonce** [pytorch/pytorch#112509]
### 🚀 The feature, motivation and pitch

There currently exists no function in the `torch.nested` namespace to create nested Tensors from buffers combined with sizes and (optionally I think) strides and storage offsets, but the C++ Code already has functions for this kind of operation: `wrap_buffer` in `aten/src/ATen/native/nested/NestedTensorUtils.h`: https://github.com/pytorch/pytorch/blob/04024926f4d6ee07bfd2368a96d848879c982314/aten/src/ATen/native/nested/NestedTensorUtils.h#L35-L45 I use nested Tensors to accelerate my Dataset storage/loading, as can be observed using the small example below, so this feature would be greatly appreciated. 

I would be willing to open an according PR if that is desired, maybe this feature can also be included for the promotion of nested Tensors to the beta stage (#112398)

```python3
import torch
from time import time

# create a random dataset of variable sequence lengths (10k elements)
sizes = torch.randint(32, 256, size=(10000,))
dataset = {
    'input_ids': [torch.randint(32000, size=(i,)) for i in sizes],
    'labels': [torch.randint(32000, size=(i,)) for i in sizes],
}

def measure(name, func):
    s = time()
    func()
    print(f"{name} took {time() - s} seconds")

# save the dataset
measure("store_dict", lambda: torch.save(dataset, "dataset.pt"))
# load the dataset
measure("load_dict", lambda: torch.load("dataset.pt", weights_only=True))

# convert the dataset to nested tensor
def to_nested_tensor_info(ds: dict):
    return {k: torch.nested.nested_tensor(v) for k, v in ds.items()}

def save_nested_tensor(ds: dict):
    nested_ds = to_nested_tensor_info(ds)
    torch.save({
        k: (
            torch.cat(v, dim=-1),  # one large tensor
            nested_ds[k]._nested_tensor_size(),
            nested_ds[k]._nested_tensor_strides(),
            nested_ds[k]._nested_tensor_storage_offsets(),
        ) for k, v in ds.items()
    }, "nested.pt")

def load_nested_tensor():
    tuple_ds = torch.load("nested.pt", weights_only=True)
    return {
        k: torch._nested_view_from_buffer(*v) for k, v in tuple_ds.items()
    }

# save the dataset
measure("store_nested", lambda: save_nested_tensor(dataset))
# load the dataset
measure("load_nested", lambda: print(load_nested_tensor()['input_ids']))

```

### Alternatives

There exists `_nested_view_from_buffer`, but it is not located in the `torch.nested.` namespace, so I am unsure about the future for the function at hand.

**Comment by jbschlosser** [comment]
Hey @fleonce, thanks for the request! I certainly agree that a public API for construction of NTs directly from a buffer and metadata is valuable, and we will consider this for inclusion during promotion to beta.

Regarding your specific example, are you only utilizing `_nested_view_from_buffer()` for serialization or are there additional use cases? Direct serialization of NTs using `torch.save()` / `torch.load()` is supported now so there should no longer be a need to manually write serialization logic. Please let me know if this doesn't work for you for some reason and we can prioritize addressing this.

**Comment by fleonce** [comment]
Hello @jbschlosser, thanks for the quick follow-up!

I am currently using the `_nested_view_from_buffer` method only for serialization. I am using the code above inside a Dataset class using the `safetensors` serialization which expects only normal Tensors with now dynamic sizes (no nested or sparse Tensors), however for my current projects I have both, dynamic sequence lengths and also sparse Tensors, both of which I have to convert to "normal" Tensor objects such that I can then store them with `safetensors`. Now thinking about it, maybe I should have included this in the example snippet.

As an aside question: is there a way to obtain the buffer for the nested Tensor as a Tensor without doing the `torch.cat(v, dim=-1)` as in the above example?
